common/download_and_verify.sh

```diff
@@ -1,29 +1,27 @@
 #!/bin/bash
 set -ex
 
-DOWNLOAD_URL=$1
-DOWNLOADED_FILE_NAME=$(basename $1)
-FILE_CHECKSUM=$2
+# Parameters
+download_url=$1
+downloaded_file_name=$(basename $1)
+file_checksum=$2
 
 # Find and verify checksum
 verify_checksum() {
     local checksum=`sha256sum $1 | awk '{print $1}'`
-    if [[ $checksum == $2 ]]
+    if [[ $checksum != $2 ]]
     then
-        echo "Checksum verified!"
-    else
         echo "*** Error - Checksum verification failed"
         exit -1
     fi
+    echo "Checksum verified!"
 }
 
-if [ $# -eq 2 ]
+if [ $# -ne 2 ]
```

Comment:
Since this script is being refactored, does it make sense to move the input arg check to the very beginning of the file?